spinalcordtoolbox/mtsat/mtsat.py

```diff
@@ -0,0 +1,139 @@
+        r1map = np.nan_to_num(r1map)
+        ind_unrealistic = np.where(r1map < 0.01)  # R1=0.01 s^-1 corresponds to T1=100s which is reasonable to clip
+        r1map[ind_unrealistic] = np.inf  # set to infinity so that these values will be 0 on the T1map
+        # compute T1
+        nii_t1map = nii_mt.copy()
+        nii_t1map.data = 1. / r1map
+    else:
+        sct.printv('Use input T1 map.', verbose)
+        r1map = 1. / nii_t1map.data
+
+    # Compute A
+    sct.printv('Compute A...', verbose)
+    a = (tr_pd * fa_t1_rad / fa_pd_rad - tr_t1 * fa_pd_rad / fa_t1_rad) * np.true_divide(np.multiply(nii_pd.data, nii_t1.data, dtype=float), tr_pd * fa_t1_rad * nii_t1.data - tr_t1 * fa_pd_rad * nii_pd.data)
+
+    # Compute MTsat
+    sct.printv('Compute MTsat...', verbose)
+    nii_mtsat = nii_mt.copy()
+    nii_mtsat.data = tr_mt * np.multiply((fa_mt_rad * np.true_divide(a, nii_mt.data) - 1), r1map, dtype=float) - (fa_mt_rad ** 2) / 2.
+    # sct.printv('nii_mtsat.data[95,89,14]' + str(nii_mtsat.data[95,89,14]), type='info')
+    # remove nans and clip unrelistic values
+    nii_mtsat.data = np.nan_to_num(nii_mtsat.data)
+    ind_unrealistic = np.where(np.abs(nii_mtsat.data) > 1)  # we expect MTsat to be on the order of 0.01
+    nii_mtsat.data[ind_unrealistic] = 0
+    # convert into percent unit (p.u.)
+    nii_mtsat.data *= 100
+
+    # Apply B1 correction to result
+    # Weiskopf, N., Suckling, J., Williams, G., Correia, M.M., Inkster, B., Tait, R., Ooi, C., Bullmore, E.T., Lutti, A., 2013. Quantitative multi-parameter mapping of R1, PD(*), MT, and R2(*) at 3T: a multi-center validation. Front. Neurosci. 7, 95.
+    if not nii_b1map is None:
+        nii_mtsat.data = np.true_divide(nii_mtsat.data * (1 - b1correctionfactor), (1 - b1correctionfactor * nii_b1map.data))
+
+    # set back old seterr settings
+    np.seterr(**seterr_old)
+
+    return nii_mtsat, nii_t1map
+
+
+def compute_mtsat_from_file(fname_mt, fname_pd, fname_t1, tr_mt, tr_pd, tr_t1, fa_mt, fa_pd, fa_t1, fname_b1map=None,
+                            fname_mtsat=None, fname_t1map=None, verbose=1):
+    """
+    Compute MTsat and T1map.
+    :param fname_mt:
+    :param fname_pd:
+    :param fname_t1:
+    :param tr_mt:
+    :param tr_pd:
+    :param tr_t1:
+    :param fa_mt:
+    :param fa_pd:
+    :param fa_t1:
+    :param fname_b1map:
+    :param fname_mtsat:
+    :param fname_t1map:
+    :param verbose:
+    :return: fname_mtsat: file name for MTsat map
+    :return: fname_t1map: file name for T1 map
+    """
+    # load data
+    sct.printv('Load data...', verbose)
+    nii_mt = Image(fname_mt)
+    nii_pd = Image(fname_pd)
+    nii_t1 = Image(fname_t1)
+    if fname_b1map is not None:
+        nii_b1map = Image(fname_b1map)
+
+    # compute MTsat
+    nii_mtsat, nii_t1map = compute_mtsat(nii_mt, nii_pd, nii_t1, tr_mt, tr_pd, tr_t1, fa_mt, fa_pd, fa_t1,
+                                         nii_b1map=None, verbose=verbose)
```

Comment:
The parameter `nii_b1map` is always `None` despite having loaded `nii_b1map` before?

Reply:
thanks, fixed in https://github.com/neuropoly/spinalcordtoolbox/pull/1672/commits/a3f9d239c2003d8c461ebf1c1f296883ceb56fb0